plutus-tx-plugin/test/AsData/Budget/Spec.hs

```diff
@@ -70,5 +77,19 @@ recordFields =
         ||]
     )
 
+recordFieldsManual :: CompiledCode (PlutusTx.BuiltinData -> Integer)
+recordFieldsManual =
+  $$( compile
+        [||
+        \d ->
+          let ints = PlutusTx.unsafeFromBuiltinData d
+              x = int1Manual ints
+              y = int2Manual ints
+              z = int3Manual ints
+              w = int4Manual ints
```

Comment:
At the moment the accessor functions are all used exactly once, which means that we're pretty much guaranteed that the inliner will inline them. And that's very important, since in order for CSE to help it has to see the inlined versions. But in reality field accessors may well be used multiple times, which might inhibit inlining (and hence CSE). So I'd like to have at least one of them used repeatedly.

Reply:
> since in order for CSE to help it has to see the inlined versions

I don't think that's true. In this particular case, I think, e.g., `unConstrData d` will be processed regardless of inlining. But I can modify the test as you suggested.

Reply:
That seems wrong. If we have two record accessors which both have `\d -> ... unConstrData d ...` then the two lambdas (correctly) create two different scopes (after all, there might be two different datas there!). The only way we can get sharing is if we inline and reduce them so that it's the _same_ d in both cases.